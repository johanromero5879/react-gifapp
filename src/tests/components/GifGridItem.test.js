import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Testing with <GifGridItem />', () => {
    const title = "A title"
    const url = "https://i.pinimg.com/originals/1c/4f/4c/1c4f4c7010a0b7ca14e32baa83fe2181.jpg"
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> )

    test('should render correclty', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('should have a paragraph with title ', () => {
        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )
    })

    test('should have a image equal to url and alt from props ', () => {
        const img = wrapper.find('img')
        
        expect( img.prop('src') ).toBe( url )
        expect( img.prop('alt') ).toBe( title )
    })

    test('should have class animate__fadeIn', () => {
        const div = wrapper.find('div')

        expect( div.hasClass('animate__fadeIn') ).toBe( true )
    })
    
    
})