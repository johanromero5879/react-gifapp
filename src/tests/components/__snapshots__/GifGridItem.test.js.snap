// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with <GifGridItem /> should render correclty 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://i.pinimg.com/originals/1c/4f/4c/1c4f4c7010a0b7ca14e32baa83fe2181.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
